src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/ChildFactory.cs

```diff
@@ -38,6 +38,11 @@ public abstract class ChildFactory
         /// </summary>
         public abstract IEnumerable<NodeFilter> Filters { get; }
 
+        /// <summary>
+        /// The list of filters that should be applied on the smo object list, concatenated by "or"
+        /// </summary>
+        public abstract IEnumerable<NodeFilter> OrFilters { get; }
```

Comment:
Please open a separate PR for adding this OR support. It will be much simpler to review both PRs if they're separated out.

But also - an OR should just be another filter. It just has two parameters, predicate1 and predicate2 (each filters themselves). So we shouldn't need to be adding a separate list of or filters here since it'll just be added to the list of Filters.

Reply:
How would this look functionally, though? Filters is a list of  NodeFilters, with each their own property, values, type, etc.

I suppose I could add an Or property, which is itself a list of NodeFilters?

Reply:
Let's get this in a separate PR and we can discuss there - it might be that the current design makes what I'm thinking harder than is worth it. I'll take a look and see what I can come up with.